### DISCUSSION TITLE
Not understanding recursive group members

### DISCUSSION THREAD

**Discussion by clevelasosu** [DirectoryTree/LdapRecord#301] [General]
I want to query an AD group and get the resulting users.  My test group contains one other group.  I'm able to get the group:

$group = Group::findBy('cn','my_group');
$members = $group->members()->recursive()->get();

If I do a foreach loop, I just get the subgroup.

foreach ($members as $member) {
    print_r($member);
}

That returns a LdapRecord\Models\ActiveDirectory\Group Object

Leaving out the ->recursive() call returns the same thing.

I just want the users.  Is that possible without a recursive loop?

**Comment by stevebauman** [comment]
Hi @clevelasosu,

Are you binding to your directory with an administrator account (I.e. an account with full domain admin)?

**Reply by clevelasosu** [reply]
No.  The user has access to the see the members of the groups.

**Reply by stevebauman** [reply]
Ok. What is your base DN set to? Is it properly set to lowest level domain component? (I.e `dc=local,dc=com`)

**Reply by clevelasosu** [reply]
Yes.  If I look at group object that is returned, it has the members in it.

**Reply by clevelasosu** [reply]
I realize this is an old thread.  I have come back to it and am still not getting where I want to be.   

Following my original example, I want to get the members of 'my_group'.  In AD, this group contains a second group "group2".  group2 contains five users.

I want to query 'my_group' and just get a list of the cn attribute for those five users.  If I run the code above, I get a single Group object.  If I print_r($members), I get a Collection that includes a single Group:


```
LdapRecord\Models\Collection Object
(
    [items:protected] => Array
        (
            [0] => LdapRecord\Models\ActiveDirectory\Group Object
                (
                    [defaultDates:protected] => Array
                        (
                            [whenchanged] => windows
                            [whencreated] => windows
                            [dscorepropagationdata] => windows
                        )

                    [sidKey:protected] => objectsid
                    [exists] => 1
                    [wasRecentlyCreated] => 
                    [wasRecentlyRenamed] => 
                    [dn:protected] => CN=group2,OU=My Groups,DC=mydomain,DC=com
                    [in:protected] => 
                    [connection:protected] => 
                    [guidKey:protected] => objectguid
                    [modifications:protected] => Array
                        (
                        )

                    [original:protected] => Array
                        (
                            [objectclass] => Array
                                (
                                    [0] => top
                                    [1] => group
                                )

                            [cn] => Array
                                (
                                    [0] => group2
                                )

                            [description] => Array
                                (
                                    [0] => group 2 users
                                )

                            [member] => Array
                                (
                                    [0] => CN=user1,OU=Accounts,DC=mydomain,DC=com
                                    [0] => CN=user2,OU=Accounts,DC=mydomain,DC=com
                                    [0] => CN=user3,OU=Accounts,DC=mydomain,DC=com
                                    [0] => CN=user4,OU=Accounts,DC=mydomain,DC=com
                                    [0] => CN=user5,OU=Accounts,DC=mydomain,DC=com
                                )
              ...

```
It may be that I just don't understand how to work with the Collection and Group objects.  In the end, I would like an array of ['user1','user2','user3','user4','user5'].  

Also related, does this function support large groups that require querying by range?  Like 10k users?

**Reply by clevelasosu** [reply]
And I do have the base_dn set to the base domain level (dc=mydomain,dc=com)
